src/services/ipfs-service.ts

```diff
@@ -110,21 +115,18 @@ export class IpfsService implements IIpfsService {
    * The record will also be pinned non-recusively.
    */
   storeRecord(record: Record<string, unknown>, options: AbortOptions = {}): Promise<CID> {
-    return this.ipfs.dag.put(
-        record,
-        { signal: options.signal, timeout: IPFS_PUT_TIMEOUT }
-    ).then(
-        // Note: While dag.put has a pin flag it always recurses and
-        // we do not want to recurse so we explicitly call pin.add.
-        (cid) => this.ipfs.pin.add(cid, {
-            signal: options.signal,
-            timeout: IPFS_PUT_TIMEOUT,
-            recursive: false
+    return this.ipfs.dag.put(record, { signal: options.signal, timeout: IPFS_PUT_TIMEOUT }).then(
```

Comment:
Do we need to rate limit writes?

Reply:
each anchor worker runs in its own process, so there's no shared memory so a semaphore won't help.  Each individual worker does writes to ipfs entirely serially, the only parallel access comes from running multiple workers in parallel.  But we only run 2 workers currently, maybe we'll bump to 3 or 4 at some point, but I'm still not too worried about the writes overloading the ipfs node.  The reads cause far more simultaneous load as there's one read for every incoming anchor request, and we get hundreds of requests a second.